pkg/controller/map/storage/handler/openstack/handler.go

```diff
@@ -92,9 +130,42 @@ func (r *Handler) changed(models ...*openstack.Storage) {
 		for _, pair := range mp.Spec.Map {
 			ref := pair.Source
 			for _, ds := range models {
-				if ref.ID == ds.ID || strings.HasSuffix(ds.Path, ref.Name) {
```

Comment:
it's unrelated to this particular PR but how about renaming `mp` and `ds` to something more meaningful?

Reply:
Done